gateway-discovery-cm/src/test/java/org/apache/knox/gateway/topology/discovery/cm/auth/AuthUtilsTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.knox.gateway.topology.discovery.cm.auth;
+
+import org.apache.knox.gateway.config.GatewayConfig;
+import org.junit.Test;
+
+import javax.security.auth.login.Configuration;
+import java.io.File;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.fail;
+
+public class AuthUtilsTest {
+
+  /**
+   * Since the login config should only be loaded once, the Configuration object should be the same for repeated
+   * requests thereof.
+   *
+   * KNOX-1962
+   */
+  @Test
+  public void testLoadJAASConfigOnce() {
+    File loginConfigFile = createTestKrb5LoginConfigFile();
+    Configuration conf1 = null;
+    Configuration conf2 = null;
+    try {
+      System.setProperty(GatewayConfig.KRB5_LOGIN_CONFIG, loginConfigFile.getAbsolutePath());
+      conf1 = AuthUtils.getKerberosJAASConfiguration();
+      conf2 = AuthUtils.getKerberosJAASConfiguration();
+    } catch (Throwable e) {
+      e.printStackTrace();
+      fail(e.getMessage());
```

Comment:
should be able to just `fail(e)` which prints the stacktrace

Reply:
Yes, that got left in when I was debugging my initial bad config file contents. It should go away.